DscResources/MSFT_xMySqlGrant/MSFT_xMySqlGrant.psm1

```diff
@@ -22,236 +15,166 @@ GrantDoesNotExist=A user with the name {0} does not exist.
 
 Import-LocalizedData LocalizedData -FileName MSFT_xMySqlGrant.strings.psd1
 
-#########################################################################################################################################
-# Set-MySqlPwdEnvironmentVariable ([pscredential] $RootPassword). Given the input root password, set the MySQL password environment variable
-#########################################################################################################################################
-
-function Set-MySqlPwdEnvironmentVariable
-{
-    param
-    (
-        [pscredential] $RootPassword
-    )
-    Write-Verbose "Setting MySql Server root password to: $($RootPassword.GetNetworkCredential().Password)"
-    [System.Environment]::SetEnvironmentVariable("MySql_PWD","$($RootPassword.GetNetworkCredential().Password)")
-}
-
-
-#########################################################################################################################################
-# Get-TargetResource ([string]$UserName, [string]DatabaseName, [string]Ensure, [string]ConnectionCredential, [string]$PermissionType) : given the username
-# determine what grants/permissions are given in which database and tables
-#########################################################################################################################################
+$ErrorPath = Join-Path -Path "$env:Temp" -ChildPath "MySQLErrors.txt"
 
 function Get-TargetResource
 {
     [OutputType([Hashtable])]
     param
     (
-       [parameter(Mandatory = $true)]
+        [parameter(Mandatory = $true)]
         [ValidateNotNullOrEmpty()]
-        [System.String]
-        $UserName,
+        [string] $UserName,
 
         [parameter(Mandatory = $true)]
         [ValidateNotNullOrEmpty()]
-        [System.String]
-        $DatabaseName = "*",
+        [string] $DatabaseName,
       
-        [ValidateSet("Present", "Absent")]
-        [string] $Ensure = "Present",  
-       
         [parameter(Mandatory = $true)]
-        [pscredential] $ConnectionCredential,
-
+        [pscredential] $RootCredential,
 
+        [parameter(Mandatory = $true)]
         [ValidateSet("ALL PRIVILEGES", "CREATE", "DROP", "DELETE", "INSERT", "SELECT", "UPDATE", "EXECUTE")]
-        [string] $PermissionType = "ALL PRIVILEGES"
+        [ValidateNotNullOrEmpty()]
+        [string] $PermissionType,
```

Comment:
Yes, my comment was about having a default value for this property (not defaulting it to  ALL PRIVILEGES). Please update this resource to grant the least amount of permissions (instead of making it a required property).